src/components/EmployeeGrid/EmployeeGrid.tsx

```diff
@@ -0,0 +1,116 @@
+import { useSearchParams } from "react-router-dom";
+import { useAppDispatch, useAppSelector } from "../../hooks/storeHelpers";
+import { initQueryParams } from "../../pages/ManageEmployees/constants";
+import { GridContainer } from "./EmployeeGrid.styles";
+import { getEmployeesListingData } from "../../utils";
+import EmployeeCard from "../EmployeeCard/EmployeeCard";
+import { selectRequestInProgress } from "../../core/store/requests/reducer";
+import { REQUESTS_ENUM } from "../../core/store/requests/requestsEnum";
+import { Loader } from "..";
+import { useEffect, useRef, useState } from "react";
+import { fetchEmployees } from "../../core/store/employeesList/actions";
+import { IQueryParams } from "../../interfaces/common";
+
+const EmployeeGrid = ({
+  setIsModalOpen,
+  setDeleteEmployee,
+}: {
+  setIsModalOpen: (isOpen: boolean) => void;
+  setDeleteEmployee: (deleteEmployeeId: number) => void;
+}) => {
+  const [offset, setOffset] = useState(0);
+
+  const observerTarget = useRef(null);
+  const [searchParams] = useSearchParams();
+
+  const employeeList = useAppSelector((state) => state.employees.employeesList);
+  const employeesCount = useAppSelector((state) => state.employees.count);
+  const dispatch = useAppDispatch();
+
+  let limit = 10;
+
+  const employeesFetchLoading = useAppSelector(
+    selectRequestInProgress(REQUESTS_ENUM.getEmployeesList)
+  );
+
+  const getSearchParams = (): IQueryParams => {
+    const sortBy = searchParams.get("sortBy") ?? initQueryParams.sortBy;
+    const sortDir = searchParams.get("sortDir") ?? initQueryParams.sortDir;
+    const skillIds = searchParams.get("skillIds");
+    const search = searchParams.get("search");
+    return {
+      limit,
+      offset,
+      sortBy,
+      sortDir,
+      skillIds,
+      search,
+    };
+  };
+
+  const handleLoadData = () => {
+    let hasMore = true;
+
+    if (employeesCount === undefined) hasMore = true;
+    else if (
+      (employeeList && employeeList.length >= employeesCount) ||
+      employeesCount === 0
+    ) {
+      hasMore = false;
+    }
+    if (employeesFetchLoading || !hasMore) return;
+    dispatch(fetchEmployees(getSearchParams()));
+    setOffset((prev) => prev + limit);
+  };
+
+  useEffect(() => {
+    const { current } = observerTarget;
```

Comment:
What is the use of destructuring here?

Reply:
`observerTarger.current` refers to the loader div at the bottom using useRef. The `current` is used multiple times in the useEffect, that's why it is destructured first.